core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -234,13 +250,22 @@ class LogManager(logDirs: Seq[File],
       logDirFailureChannel = logDirFailureChannel)
 
     if (logDir.getName.endsWith(Log.DeleteDirSuffix)) {
-      this.logsToBeDeleted.add(current)
+      this.logsToBeDeleted.add(log)
     } else {
-      val previous = this.logs.put(topicPartition, current)
+      val previous = {
+        if (log.isFuture)
+          this.futureLogs.put(topicPartition, log)
+        else
+          this.currentLogs.put(topicPartition, log)
+      }
       if (previous != null) {
```

Comment:
I have a question about the previous var. If the log dir name doesn't end with "-future", the `currentLogs` will put the topic partition as key and log as value, and return log. The simple logic like the below:
```
val previous = {log}
```
If the `previous` is not null, then there is an `IllegalStateException` to be thrown. right?